Remotion/Web/ClientScript/Scripts/DropDownMenu.ts

```diff
@@ -0,0 +1,928 @@
+      statusPopup.removeAttribute('role');
+      // TODO: missing nullcheck?
+      statusPopup.parentElement!.removeChild(statusPopup);
+    }
+
+    DropDownMenu._currentPopup = null;
+    DropDownMenu._currentStatusPopup = null;
+    DropDownMenu._currentMenu = null;
+  }
+
+  private static CreateItem(itemInfo: Nullable<DropDownMenu_ItemInfo>, selectionCount: number): Nullable<HTMLLIElement>
+  {
+    if (itemInfo == null)
+      return null;
+
+    var item;
+    if (itemInfo.Text == '-')
+      item = DropDownMenu.CreateSeparatorItem();
+    else
+      item = DropDownMenu.CreateTextItem(itemInfo, selectionCount);
+
+    return item;
+  }
+
+  private static CreateTextItem(itemInfo: DropDownMenu_ItemInfo, selectionCount: number): HTMLLIElement
+  {
+    var isEnabled = DropDownMenu.GetItemEnabled(itemInfo, selectionCount);
+
+    var item = document.createElement("li");
+
+    var className = DropDownMenu._itemClassName;
+    if (!isEnabled)
+      className = DropDownMenu._itemDisabledClassName;
+
+    item.className = className;
+    item.setAttribute('role', 'none');
+
+    var anchor = document.createElement("a");
+    anchor.setAttribute('role', 'menuitem');
+    anchor.setAttribute('tabindex', '-1');
+    if (isEnabled)
+    {
+      anchor.setAttribute('href', '#');
+    }
+    else
+    {
+      anchor.setAttribute('aria-disabled', 'true');
+    }
+    // TODO: missing nullcheck?
+    $(anchor).bind('click', DropDownMenu._itemClickHandler!);
+
+    item.appendChild(anchor);
+    if (isEnabled && itemInfo.Href != null)
+    {
+      var isJavaScript = itemInfo.Href.toLowerCase().indexOf('javascript:') >= 0;
+      if (isJavaScript)
+      {
+        anchor.setAttribute('javascript', itemInfo.Href);
+      }
+      else
+      {
+        var href = itemInfo.Href;
+        var target;
+        if (itemInfo.Target != null && itemInfo.Target.length > 0)
+          target = itemInfo.Target;
+        else
+          target = '_self';
+        anchor.setAttribute('javascript', 'window.open (\'' + href + '\', \'' + target + '\');');
+      }
+    }
+
+    if (itemInfo.Icon != null)
+    {
+      var icon = document.createElement('img');
+      if (isEnabled || itemInfo.IconDisabled == null)
+        icon.src = itemInfo.Icon;
+      else
+        icon.src = itemInfo.IconDisabled;
+
+      icon.className = DropDownMenu._itemIconClassName;
+      icon.style.verticalAlign = 'middle';
+      icon.alt = '';
+      icon.setAttribute('aria-hidden', 'true');
+      anchor.appendChild(icon);
+    }
+    else
+    {
+      var iconPlaceholder = document.createElement('span');
+      iconPlaceholder.className = DropDownMenu._itemIconClassName;
+      anchor.appendChild(iconPlaceholder);
+    }
+
+    if (itemInfo.DiagnosticMetadata)
+    {
+      // Do not render empty diagnostic metadata attributes
+      $.each(itemInfo.DiagnosticMetadata, function (key, value)
```

Comment:
```suggestion
      $.each(itemInfo.DiagnosticMetadata, function (key: string, value: string | boolean)
```

Otherwise they are assumed as `any` because the jQuery type annotations.